### PR TITLE
nkushnir oop + collections task

### PR DESCRIPTION
**Pull request by 51stRoR**
Notes:

1. As a base List used list of products(Food  class objects) from Menu object, so all methods to show, filter, check is in Menu class and called by Menu object
2. Result of filtering food items stored in additional filtered list in Menu object to be able to work with base product list in the following flow
3. Added simple console menu to test logic according to homework tasks
This menu has no validation for input values on theirs type and no validation for option choices, so DO NOT try to input values of other type than asked or select options out of suggested choices to avoid crashing
4. Last task about traversing and collect lowest price per dish category done with 2 additional temp Lists inside method - this is my first approach to solve this for now, may be redone later in a more better way
5. Added example map to Menu class to store [type of Food : amount for this type in menu lits]

### CODE REVIEW

File: src/main/java/com/burgerking/Cookable.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.burgerking;
+
+public interface Cookable {
+
+    public void cooking();
+}
```
Comment (alexyablonskyi) [inline src/main/java/com/burgerking/Cookable.java:6]: All methods in Interface is public. No needs to add `public`.

File: src/main/java/com/burgerking/FoodType.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.burgerking;
+
+public enum FoodType {
+    BURGER, BEVERAGE, COFFEE, FRIED, SALAD;
+
+    static final FoodType[] foodTypes = FoodType.values();
+}
```
Comment (alexyablonskyi) [inline src/main/java/com/burgerking/FoodType.java:5]: Why do you need `foodTypes` variable? You can any time call values() method from Enum.

File: src/main/java/com/burgerking/Menu.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    public void filter(int filterOption) {
+        List<Food> filtered = new ArrayList<>();
+        for(Food food: this.getProductList()){
+            if(food.getType().ordinal()==filterOption){
+                filtered.add(food);
+            }
+        }
+        this.setFilteredList(filtered);
+    }
+
```
Comment (alexyablonskyi) [inline src/main/java/com/burgerking/Menu.java:100]: There are a lot of unused methods in class. Do we need all of them now?

File: src/main/java/com/burgerking/Menu.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+    public boolean isOneOfBigger(Integer calories){
+        for(Food food: this.getProductList()){
+            if(food.getCalories()>calories){
+                return true;
+            }
+        }
+        return false;
+    }
+
```
Comment (alexyablonskyi) [inline src/main/java/com/burgerking/Menu.java]: `isOneOfBigger` method name is unclear. It is very difficult to understand what is method doing. Java convention says method name should represent logic of method body. This remark concern to most of methods in this class.

**Review by alexyablonskyi — COMMENTED** [review]
Please correct all remarks and you can merge your changes.

File: src/main/java/com/burgerking/Cookable.java (hunk 1 above)
Comment (51stRoR) [reply src/main/java/com/burgerking/Cookable.java:6]: modifier removed

File: src/main/java/com/burgerking/FoodType.java (hunk 2 above)
Comment (51stRoR) [reply src/main/java/com/burgerking/FoodType.java:5]: unused array removed

File: src/main/java/com/burgerking/Menu.java (hunk 3 above)
Comment (51stRoR) [reply src/main/java/com/burgerking/Menu.java:100]: Initial idea was to have 3 methods  -  to check if at least one item from list, all items from list, none of the item from list - overloaded for price, calories and size, that's why there is 9 methods with unclear names
However, its not a best idea and indeed only 3of them is needed right now = unused removed

File: src/main/java/com/burgerking/Menu.java (hunk 4 above)
Comment (51stRoR) [reply src/main/java/com/burgerking/Menu.java]: change names of check by criteria and filter() and display() methods to more clear one

**Pull request merged**